hrm-domain/hrm-service/src/permissions/canPerformActionOnObject.ts

```diff
@@ -51,7 +51,7 @@ export const canPerformActionsOnObject = async <T extends TargetKind>({
 
     switch (targetKind) {
       case 'contact': {
-        const object = await getContactById(accountSid, objectId);
+        const object = await getContactById(accountSid, objectId, { can, user });
```

Comment:
The "double dipping" into the permissions system that using the controller level access in these initial permissions checks causes is probably not a great long term solution. We've already talked about using custom "security information" db level queries for these as part of the the permissions work. I emulated the current cases behavior to minimize the impact of this PR, but we probably want to use unauthenticated/untransformed queries for these checks that are limited to the relevant data instead of accessing controller level object getters in the permission layer in the near future.

Reply:
There will be users who don't have "ViewContact" or "ViewCase" permission, but needs to pass a permission check in canPerformActionOnObject in the future (if there aren't already). Using the controller function to load the contact or case which validates view permissions means that checking permissions for those users will fail because they don't have view permission for the object, but they may have other permissions. We should probably use the db-access layer to load a limited version of the object with only the properties needed by the permission check in this file to avoid that problem in the future.